src/daemon/input_linux.c

```diff
@@ -53,6 +53,9 @@ int uinputopen(struct uinput_user_dev* indev, int mouse){
 ///
 /// Some tips on using [uinput_user_dev in](http://thiemonge.org/getting-started-with-uinput)
 int os_inputopen(usbdevice* kb){
+    // Don't create an input device if no HID input
+    if (IS_SINGLE_EP(kb))
+        return 1;
```

Comment:
You need to zero the file descriptors before bailing, so that os_inputclose bails when called.